core/src/main/scala/kafka/coordinator/transaction/TransactionMetadata.scala

```diff
@@ -375,7 +375,7 @@ private[transaction] class TransactionMetadata(val transactionalId: String,
                                   newTopicPartitions: immutable.Set[TopicPartition],
                                   newTxnStartTimestamp: Long,
                                   updateTimestamp: Long): TxnTransitMetadata = {
-    if (pendingState.isDefined)
+    if (pendingState.isDefined && pendingState.get != newState)
```

Comment:
Without this change we fail to update since we mark pending state CompleteCommit/Abort multiple times. I need to see if there is any problem with this change. For this specific case, I think it is fine because we are pending the same state.

Reply:
nit: `pendingState.exists(_ != newState)`?